### PR TITLE
Category moderation

### LINKED ISSUE #577: Category names should be moderated

### PR DESCRIPTION
**Pull request by ritmatter**
Fixes #577

Lemme merge because we need a new env variable

### CODE REVIEW

File: api/models/Notification.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ var Schema = mongoose.Schema;
 var NotificationSchema = new Schema({
   user      : {
     id      : { type: String, required: true },
-    name    : { type: String, required: true },
+    name    : { type: String },
```
Comment (ritmatter) [inline api/models/Notification.js]: This field is never used and should not be necessary

File: api/routes/UserRoutes.js [hunk 2]
```diff
@@ -135,7 +136,7 @@ module.exports = function(router, isAuthenticated, acl, censor) {
           winston.log('error', 'Error creating user: %s', msg);
           return res.status(501).send(msg);
         } else {
-          var verificationString = utils.generateVerificationToken();
+          var verificationString = utils.getRandomString();
```
Comment (ritmatter) [inline api/routes/UserRoutes.js]: Just some name changes around here

File: config/acl.js [hunk 3]
```diff
@@ -3,14 +3,12 @@ var whitelist = require('./whitelist.js');
 
 // Express middleware to check for acl privileges based on a whitelist
 var acl = {
-
   // Check if the user has admin privileges
   isAdmin: function(req, res, next) {
```
Comment (ritmatter) [inline config/acl.js]: New (definitely working) middleware to detect an admin. Requires a password in the body

**Comment by krazemon** [comment]
LGTM, this is cool!

**Pull request merged**